python-lib/sharepoint_lists.py

```diff
@@ -78,15 +75,39 @@ def flush(self):
         self.list_item_entity_type_full_name = created_list.get("ListItemEntityTypeFullName")
         logger.info('New list "{}" created, type {}'.format(self.list_item_entity_type_full_name, self.entity_type_name))
         self.list_id = created_list.get("Id")
+        self.max_workers = max_workers
+        self.batch_size = batch_size
+        self.working_batch_size = max_workers * batch_size
 
+    def write_row(self, row):
+        self.buffer.append(row)
+        if len(self.buffer) >= self.working_batch_size:
+            self.flush()
+            self.buffer = []
+
+    def flush(self):
         self.parent.get_read_schema()
         self.create_sharepoint_columns()
 
         logger.info("Starting adding rows")
-        for row in self.buffer:
-            item = self.build_row_dictionary(row)
-            self.parent.client.add_list_item_by_id(self.list_id, self.list_item_entity_type_full_name, item)
-        logger.info("All rows added")
+        index = 0
+        offset = 0
+        kwargs = []
+        futures = []
+        with ThreadPoolExecutor(max_workers=self.max_workers) as thread_pool_executor:
+            for row in self.buffer:
```

Comment:
I disagree with this method as the input dataset order is not kept.
If you have 5 chunks of rows to send, each identified by an id (so, chunk1 is [0..99], ..., chunk5 [400...499]
The exported list will be:
- id:0
- id:100
- id:200
- id:300
- id:400
- id:1
- id:101
...

Reply:
The multi threaded method is now disabled by default, and a warning message regarding items order as been added.